sdk/core/azure-core/azure/core/tracing/ext/opencensus_wrapper.py

```diff
@@ -0,0 +1,169 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+
+from azure.core.tracing import AbstractSpan
+from opencensus.trace import execution_context
+from opencensus.trace import tracer as tracer_module, Span
+from opencensus.trace.propagation import trace_context_http_header_format
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracers.noop_tracer import NoopTracer
+try:
+    from opencensus.ext.azure.trace_exporter import AzureExporter
+except:
+    AzureExporter = False
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Union
+
+
+class OpencensusWrapper(AbstractSpan):
+    """Wraps a given Opencensus Span so that it implements azure.core.tracing.AbstractSpan"""
+
+    def __init__(self, span=None, name="parent_span"):
+        # type: (Span, str) -> None
+        """
+        If a span is not passed in, creates a new tracer. If the instrumentation key for Azure Exporter is given, will
+        configure the azure exporter else will just create a new tracer.
+
+        :param span: The opencensus span to wrap
+        :param name: The name of the opencensus span to create if a new span is needed
+        """
+        super(OpencensusWrapper, self).__init__(span, name)
+        tracer = self.get_current_tracer()
+        self.was_created_by_azure_sdk = False
+        if span is None and (tracer is None or isinstance(tracer, NoopTracer)):
+            azure_exporter_instrumentation_key = self._get_environ(
+                "APPINSIGHTS_INSTRUMENTATIONKEY"
+            )
+            prob = float(self._get_environ("AZURE_TRACING_SAMPLER") or 0)
+            if azure_exporter_instrumentation_key is not None and AzureExporter:
```

Comment:
That means the user purposefully set the value to an empty value also `bool('')` is False in python.

Reply:
Yes, what I'm getting at is whether we want to go ahead with an empty string, because it's invalid. `AzureExporter` raises `ValueError` in that case, so I think we do.

Reply:
I deleted this whole section after talking with azure monitering